### PR TITLE
add windows ci

### PR DESCRIPTION
**Pull request by gengjiawen**
## Motivation
[Build status](https://ci.appveyor.com/project/gengjiawen/react-native/branch/master)
Add windows ci
## Test Plan
pass all current ci.
## Related PRs
none
## Release Notes
 [GENERAL] [INTERNAL] [CI] - add windows ci

### CODE REVIEW

**Comment by gengjiawen** [comment]
cc @hramos

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    - yes 2> nul | sdkmanager --licenses > nul
+    - sdkmanager "system-images;android-19;google_apis;armeabi-v7a"
+    - sdkmanager "platforms;android-26"
+    - sdkmanager  "build-tools;26.0.3"
+    - sdkmanager "add-ons;addon-google_apis-google-23"
+    - sdkmanager "extras;android;m2repository"
+
+    - ps: Install-Product node $env:nodejs_version
+    - npm install
```
Comment (matthargett) [inline appveyor.yml]: I suggest doing `appveyor-retry npm` to make it resilient against intermittent network issues. I added this in the react-native-windows AppVeyor configuration to eliminate a not-uncommon source of flakiness. https://github.com/Microsoft/react-native-windows/blob/master/appveyor.yml#L40

You may also want to add `appveyor-retry npm install npm@lts`.

**Comment by matthargett** [comment]
Right now, the build doesn't get to `npm test` -- it fails on the gradle command, with the failure indicator saying `:RNTester:android:app:bundleReleaseJsAndAssets FAILED`.

Is that what we expect? Can we up the logging level so the error is more detailed and easier for users without a local Windows instance to reason about?

**Comment by matthargett** [comment]
I love how clean this is with things extracted into variables, btw. Great job! 🥇

**Comment by gengjiawen** [comment]
The failure is expected, since the windows issue still exists. Hopefully @rafeca can fix https://github.com/facebook/metro/issues/181 today. The gradle error message in most case in sufficient.

**Comment by gengjiawen** [comment]
@hramos Can you enable appveyor for react native, I want to add the badge in README.md.

**Comment by hramos** [comment]
Enabled.

**Comment by gengjiawen** [comment]
@hramos I have updated README.md. Do you plan to merge this or after have the windows build fixed ?

**Pull request closed without merging**